### ISSUE TITLE
Katib argo workflow creation failed

### ISSUE THREAD

**Issue by md-shabbir** [argoproj/argo#1378]
**Is this a BUG REPORT or FEATURE REQUEST?**:
FEATURE REQUEST

**What happened**:
Katib spec is not working with argo workflow.
```yaml
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  name: test
  namespace: kubeflow
spec:
  serviceAccountName: studyjob-controller
  entrypoint: start
  templates:
  - name: start
    resource:                   # indicates that this is a resource template
      action: create            # can be any kubectl action (e.g. create, delete, apply, patch)
      manifest: |               #put your kubernetes spec here
        apiVersion: "kubeflow.org/v1alpha1"
        kind: StudyJob
        metadata:
          namespace: kubeflow
          labels:
            controller-tools.k8s.io: "1.0"
          name: tfjob-example
        spec:
          studyName: tfjob-example
          owner: crd
          optimizationtype: maximize
          objectivevaluename: accuracy_1
          optimizationgoal: 0.99
          requestcount: 4
          metricsnames:
            - accuracy_1
          parameterconfigs:
            - name: --learning_rate
              parametertype: double
              feasible:
                min: "0.01"
                max: "0.05"
            - name: --batch_size
              parametertype: int
              feasible:
                min: "100"
                max: "200"
          workerSpec:
            goTemplate:
                rawTemplate: |-
                  apiVersion: "kubeflow.org/v1beta1"
                  kind: TFJob
                  metadata:
                    name: {{.WorkerID}}
                    namespace: kubeflow
                  spec:
                   tfReplicaSpecs:
                    Worker:
                      replicas: 1
                      restartPolicy: Never
                      template:
                        spec:
                          containers:
                            - name: tensorflow
                              image: gcr.io/kubeflow-ci/tf-mnist-with-summaries:1.0
                              command:
                                - "python"
                                - "/var/tf_mnist/mnist_with_summaries.py"
                                - "--log_dir=/train/{{.WorkerID}}"
                                {{- with .HyperParameters}}
                                {{- range .}}
                                - "{{.Name}}={{.Value}}"
                                {{- end}}
                                {{- end}}
                              volumeMounts:
                                - mountPath: "/train"
                                  name: "train"
                          volumes:
                            - name: "train"
                              persistentVolumeClaim:
                                claimName: "tfevent-volume"
          metricsCollectorSpec:
            goTemplate:
              rawTemplate: |-
                apiVersion: batch/v1beta1
                kind: CronJob
                metadata:
                  name: {{.WorkerID}}
                  namespace: kubeflow
                spec:
                  schedule: "*/1 * * * *"
                  successfulJobsHistoryLimit: 0
                  failedJobsHistoryLimit: 1
                  jobTemplate:
                    spec:
                      template:
                        spec:
                          containers:
                          - name: {{.WorkerID}}
                            image: gcr.io/kubeflow-ci/katib/tfevent-metrics-collector:v0.1.2-alpha-77-g9324cad
                            args:
                            - "python"
                            - "main.py"
                            - "-m"
                            - "vizier-core"
                            - "-s"
                            - "{{.StudyID}}"
                            - "-w"
                            - "{{.WorkerID}}"
                            - "-d"
                            - "/train/{{.WorkerID}}"
                            volumeMounts:
                                - mountPath: "/train"
                                  name: "train"
                          volumes:
                            - name: "train"
                              persistentVolumeClaim:
                                  claimName: "tfevent-volume"
                          restartPolicy: Never
                          serviceAccountName: metrics-collector
          suggestionSpec:
            suggestionAlgorithm: "random"
            requestNumber: 3
```
When I submit this yaml, I am getting following error
```
2019/05/28 11:01:29 Failed to submit workflow: templates.start: failed to resolve {{.WorkerID}}
```
I know argo is tryng to resolve it and fails. But I want argo to not resolve it.

**What you expected to happen**:
I want to escape "{{" and "}}".

**How to reproduce it (as minimally and precisely as possible)**:
Run the given yaml.

**Anything else we need to know?**:
No

**Environment**:
- Argo version:
```
$ argo version
argo: v2.2.1
  BuildDate: 2018-10-11T16:26:28Z
  GitCommit: 3b52b26190163d1f72f3aef1a39f9f291378dafb
  GitTreeState: clean
  GitTag: v2.2.1
  GoVersion: go1.10.3
  Compiler: gc
  Platform: linux/amd64
```
- Kubernetes version :
```
$ kubectl version -o yaml
clientVersion:
  buildDate: 2018-10-24T06:54:59Z
  compiler: gc
  gitCommit: 17c77c7898218073f14c8d573582e8d2313dc740
  gitTreeState: clean
  gitVersion: v1.12.2
  goVersion: go1.10.4
  major: "1"
  minor: "12"
  platform: linux/amd64
serverVersion:
  buildDate: 2019-01-16T18:14:49Z
  compiler: gc
  gitCommit: 51dd616cdd25d6ee22c83a858773b607328a18ec
  gitTreeState: clean
  gitVersion: v1.12.5
  goVersion: go1.10.7
  major: "1"
  minor: "12"
  platform: linux/amd64
```

**Comment by elikatsis** [comment]
Try upgrading your Argo installation to v2.3.0.
I think this issue is resolved by #1283.

**Comment by md-shabbir** [comment]
> Try upgrading your Argo installation to v2.3.0.
> I think this issue is resolved by #1283.

I upgraded to v2.3.0 and it did the job.
Thanks @elikatsis
